src/world_diff.rs

```diff
@@ -26,11 +27,13 @@ pub struct WorldDiff {
 
     // The fields below are only rolled back when the whole VM is rolled back.
     pub(crate) decommitted_hashes: RollbackableSet<U256>,
-    read_storage_slots: RollbackableSet<(H160, U256)>,
-    written_storage_slots: RollbackableSet<(H160, U256)>,
+    pub read_storage_slots: RollbackableSet<(H160, U256)>,
+    pub written_storage_slots: RollbackableSet<(H160, U256)>,
 
     // This is never rolled back. It is just a cache to avoid asking these from DB every time.
     storage_initial_values: BTreeMap<(H160, U256), Option<U256>>,
+
+    pub storage_application_cycles: u32,
```

Comment:
If we want to retain stats in this repo, I'd much rather maintain separate read and write stats so that they can be easily aggregated by the caller. Also, I *really* don't like exposing this data as public writeable fields. IMO, interface like `take_storage_cycles(&mut self) -> (u32, u32)` resetting the stats to zero would be preferable (same for precompile stats if they are retained inside the VM).